src/Orleans.Core/Threading/ThreadPoolExecutor.cs

```diff
@@ -384,71 +401,78 @@ internal class ThreadPoolTrackingStatistic
         private readonly ThreadTrackingStatistic threadTracking;
 
         private readonly QueueTrackingStatistic queueTracking;
+        private readonly bool collectDetailedThreadStats;
+        private readonly bool collectQueueStats;
+        private readonly bool collectDetailedQueueStats;
 
-        public ThreadPoolTrackingStatistic(string name, ILoggerFactory loggerFactory)
+        public ThreadPoolTrackingStatistic(string name, ILoggerFactory loggerFactory, IOptions<StatisticsOptions> statisticsOptions, StageAnalysisStatisticsGroup schedulerStageStatistics)
         {
-            if (StatisticsCollector.CollectQueueStats)
+            var collectionLevel = statisticsOptions.Value.CollectionLevel;
+            this.collectQueueStats = collectionLevel >= StatisticsLevel.Verbose2;
+            if (this.collectQueueStats)
             {
-                queueTracking = new QueueTrackingStatistic(name);
+                queueTracking = new QueueTrackingStatistic(name, statisticsOptions);
             }
 
-            if (ExecutorOptions.CollectDetailedThreadStatistics)
+            this.collectDetailedQueueStats = StatisticsOptions.TRACK_DETAILED_STATS && this.collectQueueStats;
+            this.collectDetailedThreadStats = StatisticsOptions.TRACK_DETAILED_STATS && collectionLevel >= StatisticsLevel.Verbose2;
```

Comment:
I think this logic should be in `ThreadTrackingStatistic`. The logic on whether to collect a set of statistics should be in the statisticGroup itself